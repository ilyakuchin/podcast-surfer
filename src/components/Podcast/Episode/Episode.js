import React, { Component } from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";

const EpisodeGrid = styled.li`
  width: 100%;
  margin-top: 10px;
  margin-bottom: 10px;
  display: grid;
  grid-gap: 15px;
  grid-template-areas: "episode-image episode-link";
  justify-items: center;
  align-items: center;
  grid-template-columns: 200px auto;

  @media (max-width: 600px) {
    margin-top: 50px;
    margin-bottom: 50px;
    grid-template-columns: auto;
    grid-template-areas:
      "episode-image"
      "episode-link";
  }
`;

const EpisodeImage = styled.img`
  grid-area: episode-image;
  width: 200px;
  height: 200px;
  object-fit: scale-down;
`;

const EpisodeLink = styled.div`
  max-width: 700px;
  grid-area: episode-link;

  @media (max-width: 600px) {
    margin-left: 10px;
    margin-right: 10px;
    text-align: center;
  }
`;

export default class Episode extends Component {
  render() {
    return (
      <EpisodeGrid>
        <EpisodeImage src={this.props.image} alt="episode cover" />
        <EpisodeLink>
          <Link
            to={{
              pathname: "/episode-player",
              state: {
                name: this.props.name,
                description: this.props.description,
                image: this.props.image,
                audio: this.props.audio
              }
            }}
          >
            {this.props.name}
          </Link>
        </EpisodeLink>
      </EpisodeGrid>
    );
  }
}
