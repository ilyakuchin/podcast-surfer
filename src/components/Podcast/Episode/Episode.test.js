import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Episode from "./Episode";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("Test Episode Component", () => {
  test("should render correctly", () => {
    expect(toJson(shallow(<Episode />))).toMatchSnapshot();
  });
});
