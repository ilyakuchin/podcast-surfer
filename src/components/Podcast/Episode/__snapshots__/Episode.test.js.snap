// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Episode Component should render correctly 1`] = `
<styled.li>
  <styled.img
    alt="episode cover"
  />
  <styled.div>
    <Link
      to={
        Object {
          "pathname": "/episode-player",
          "state": Object {
            "audio": undefined,
            "description": undefined,
            "image": undefined,
            "name": undefined,
          },
        }
      }
    />
  </styled.div>
</styled.li>
`;
